From fa4318473a8da12b9a4d21c1530269da3ff56637 Mon Sep 17 00:00:00 2001
From: Bastian Rihm <me@bastianjoel.de>
Date: Fri, 25 May 2018 23:51:33 +0200
Subject: [PATCH] loadCss return Promise

---
 nativescript-fonticon.ts | 27 +++++++++++++++++----------
 1 file changed, 17 insertions(+), 10 deletions(-)

diff --git a/nativescript-fonticon.ts b/nativescript-fonticon.ts
index 025d7f5..cd2240c 100755
--- a/nativescript-fonticon.ts
+++ b/nativescript-fonticon.ts
@@ -6,7 +6,7 @@ export class TNSFontIcon {
   public static paths: any = {}; // file paths to font icon collections
   public static debug: boolean = false;
 
-  public static loadCss(): void {
+  public static loadCss(): Promise<any> {
     let cnt = 0;
     let currentName: string;
     let fontIconCollections = Object.keys(TNSFontIcon.paths);
@@ -59,17 +59,24 @@ export class TNSFontIcon {
       });
     };    
     
-    let loadFiles = () => {
-      initCollection();
-      if (cnt < fontIconCollections.length) {
-        loadFile(TNSFontIcon.paths[currentName]).then(() => {
-          cnt++;
-          loadFiles();
-        });
-      }
+    let loadFiles = (): Promise<any> => {
+      return new Promise((resolve) => {
+        initCollection();
+        
+        if (cnt < fontIconCollections.length) {
+          loadFile(TNSFontIcon.paths[currentName]).then(() => {
+            cnt++;
+            return loadFiles().then(() => {
+              resolve()
+            });
+          });
+        } else {
+          resolve();
+        }
+      });
     };
 
-    loadFiles();         
+    return loadFiles();         
   }
 }
 
